core/src/main/java/com/graphhopper/routing/util/NameSimilarityEdgeFilter.java

```diff
@@ -103,33 +105,42 @@ String getNormalizedPointHint() {
      * TODO Currently limited to certain 'western' languages
      */
     private String prepareName(String name) {
-        // \s = A whitespace character: [ \t\n\x0B\f\r]
-        String[] arr = name.split("\\s");
-        List<String> list = new ArrayList<>(arr.length);
-        for (int i = 0; i < arr.length; i++) {
-            String rewrite = NON_WORD_CHAR.matcher(toLowerCase(arr[i])).replaceAll("");
-            String tmp = rewriteMap.get(rewrite);
-            if (tmp != null)
-                rewrite = tmp;
-            boolean isNumber = false;
-            try {
-                if (Integer.parseInt(rewrite) > 0)
-                    isNumber = true;
-            } catch (NumberFormatException ex) {
+        final char separator = ' ';
+        int lastMatch = 0, nextMatch;
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i <= name.length(); i++) {
+            // string.split("\\s") can be slow. prepareName is called often so we make it fast although a bit unreadable also avoid trim().
+            // another bottleneck is NON_WORD_CHAR.replaceAll but it is harder to replace. Or we could create a cache
+            if (i < name.length() && name.charAt(i) != separator)
+                continue;
+            if (lastMatch == i) {
+                lastMatch++;
+                continue;
             }
-            // Ignore matching short frases like de, la, ... except it is a number
-            if (rewrite.length() > 2 || isNumber) {
-                list.add(rewrite);
+            String token = name.substring(lastMatch, i);
+            lastMatch = i + 1;
+            // end string.split
```

Comment:
Will eventually revert this split complexity as it seems it does not improve speed.